src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,52 @@
+package duke.task;
+
+/**
+ * Class representing a generic Task
+ */
+public class Task {
```

Comment:
Since Task is never instantiated directly, you should make it an abstract class and allow subclasses to have their own implementation of the methods instead of overriding every method when Task is never instantiated to begin with